File: hylife/simulation/equilibrium_MHD.py
from pyccel.decorators import types
import ..geometry.mappings_analytical as mapping




# ============================ physical domain ===========================================================
# equilibrium bulk pressure (x - component)
@types('double','double','double')
def p_eq_phys(x, y, z):
    
    #p_phys = 1.
    p_phys = (1 - x) * (1 - y) * (1 - z) * x * y * z
    
    return p_phys

# equilibrium bulk velocity (x - component)
@types('double','double','double')
def ux_eq(x, y, z):
    
    ux = 0.
    #ux = cos(2*pi*xi1) * sin(2*pi*xi2) * sin(2*pi*xi3) * 2*pi
    
    return ux

# equilibrium velocity (y - component)
@types('double','double','double')
def uy_eq(x, y, z):
    
    uy = 0.
    #uy = sin(2*pi*xi1) * cos(2*pi*xi2) * sin(2*pi*xi3) * 2*pi
    
    return uy

# equilibrium bulk velocity (z - component)
@types('double','double','double')
def uz_eq(x, y, z):
    
    uz = 0.
    #uz = sin(2*pi*xi1) * sin(2*pi*xi2) * cos(2*pi*xi3) * 2*pi
    
    return uz

# equilibrium magnetic field (x - component)
@types('double','double','double')
def bx_eq(x, y, z):
    
    bx = 1.
    #bx = sin(2*pi*xi1) * cos(2*pi*xi2) * cos(2*pi*xi3) * (2*pi)**2
    
    return bx

# equilibrium magnetic field (y - component)
@types('double','double','double')
def by_eq(x, y, z):
    
    by = 0.
    #by = cos(2*pi*xi1) * sin(2*pi*xi2) * cos(2*pi*xi3) * (2*pi)**2
    
    return by

# equilibrium magnetic field (z - component)
@types('double','double','double')
def bz_eq(x, y, z):
    
    bz = 0.
    #bz = cos(2*pi*xi1) * cos(2*pi*xi2) * sin(2*pi*xi3) * (2*pi)**2
    
    return bz

# equilibrium bulk density
@types('double','double','double')
def rho_eq_phys(x, y, z):
    
    rho_phys = 1.
    #rho_phys = cos(2*pi*xi1) * cos(2*pi*xi2) * cos(2*pi*xi3) * (2*pi)**3
    
    return rho_phys







# ============================ logical domain ===========================================================
# equilibrium bulk pressure (0-form on logical domain)
@types('double','double','double','int','double[:]')
def p_eq(xi1, xi2, xi3, kind, params):
    
    x = mapping.f(xi1, xi2, xi3, kind, params, 1)
    y = mapping.f(xi1, xi2, xi3, kind, params, 2)
    z = mapping.f(xi1, xi2, xi3, kind, params, 3)
    
    p_phys = p_eq_phys(x, y, z)
    
    return p_phys

# equilibrium bulk velocity (1-form on logical domain, 1 - component)
@types('double','double','double','int','double[:]')
def u1_eq(xi1, xi2, xi3, kind, params):
    
    x = mapping.f(xi1, xi2, xi3, kind, params, 1)
    y = mapping.f(xi1, xi2, xi3, kind, params, 2)
    z = mapping.f(xi1, xi2, xi3, kind, params, 3)
    
    df_11 = mapping.df(xi1, xi2, xi3, kind, params, 11)
    df_21 = mapping.df(xi1, xi2, xi3, kind, params, 21)
    df_31 = mapping.df(xi1, xi2, xi3, kind, params, 31)
    
    ux = ux_eq(x, y, z)
    uy = uy_eq(x, y, z)
    uz = uz_eq(x, y, z)
    
    return df_11 * ux + df_21 * uy + df_31 * uz

# equilibrium bulk velocity (1-form on logical domain, 2 - component)
@types('double','double','double','int','double[:]')
def u2_eq(xi1, xi2, xi3, kind, params):
    
    x = mapping.f(xi1, xi2, xi3, kind, params, 1)
    y = mapping.f(xi1, xi2, xi3, kind, params, 2)
    z = mapping.f(xi1, xi2, xi3, kind, params, 3)
    
    df_12 = mapping.df(xi1, xi2, xi3, kind, params, 12)
    df_22 = mapping.df(xi1, xi2, xi3, kind, params, 22)
    df_32 = mapping.df(xi1, xi2, xi3, kind, params, 32)
    
    ux = ux_eq(x, y, z)
    uy = uy_eq(x, y, z)
    uz = uz_eq(x, y, z)
    
    return df_12 * ux + df_22 * uy + df_32 * uz

# equilibrium bulk velocity (1-form on logical domain, 3 - component)
@types('double','double','double','int','double[:]')
def u3_eq(xi1, xi2, xi3, kind, params):
    
    x = mapping.f(xi1, xi2, xi3, kind, params, 1)
    y = mapping.f(xi1, xi2, xi3, kind, params, 2)
    z = mapping.f(xi1, xi2, xi3, kind, params, 3)
    
    df_13 = mapping.df(xi1, xi2, xi3, kind, params, 13)
    df_23 = mapping.df(xi1, xi2, xi3, kind, params, 23)
    df_33 = mapping.df(xi1, xi2, xi3, kind, params, 33)
    
    ux = ux_eq(x, y, z)
    uy = uy_eq(x, y, z)
    uz = uz_eq(x, y, z)
    
    return df_13 * ux + df_23 * uy + df_33 * uz

# equilibrium magnetic field (2-form on logical domain, 1 - component)
@types('double','double','double','int','double[:]')
def b1_eq(xi1, xi2, xi3, kind, params):
    
    x = mapping.f(xi1, xi2, xi3, kind, params, 1)
    y = mapping.f(xi1, xi2, xi3, kind, params, 2)
    z = mapping.f(xi1, xi2, xi3, kind, params, 3)
    
    dfinv_11 = mapping.df_inv(xi1, xi2, xi3, kind, params, 11)
    dfinv_12 = mapping.df_inv(xi1, xi2, xi3, kind, params, 12)
    dfinv_13 = mapping.df_inv(xi1, xi2, xi3, kind, params, 13)
    
    det_df   = mapping.det_df(xi1, xi2, xi3, kind, params)
    
    bx = bx_eq(x, y, z)
    by = by_eq(x, y, z)
    bz = bz_eq(x, y, z)
    
    return (dfinv_11 * bx + dfinv_12 * by + dfinv_13 * bz) * det_df

# equilibrium magnetic field (2-form on logical domain, 2 - component)
@types('double','double','double','int','double[:]')
def b2_eq(xi1, xi2, xi3, kind, params):
    
    x = mapping.f(xi1, xi2, xi3, kind, params, 1)
    y = mapping.f(xi1, xi2, xi3, kind, params, 2)
    z = mapping.f(xi1, xi2, xi3, kind, params, 3)
    
    dfinv_21 = mapping.df_inv(xi1, xi2, xi3, kind, params, 21)
    dfinv_22 = mapping.df_inv(xi1, xi2, xi3, kind, params, 22)
    dfinv_23 = mapping.df_inv(xi1, xi2, xi3, kind, params, 23)
    
    det_df   = mapping.det_df(xi1, xi2, xi3, kind, params)
    
    bx = bx_eq(x, y, z)
    by = by_eq(x, y, z)
    bz = bz_eq(x, y, z)
    
    return (dfinv_21 * bx + dfinv_22 * by + dfinv_23 * bz) * det_df

# equilibrium magnetic field (3-form on logical domain, 3 - component)
@types('double','double','double','int','double[:]')
def b3_eq(xi1, xi2, xi3, kind, params):
    
    x = mapping.f(xi1, xi2, xi3, kind, params, 1)
    y = mapping.f(xi1, xi2, xi3, kind, params, 2)
    z = mapping.f(xi1, xi2, xi3, kind, params, 3)
    
    dfinv_31 = mapping.df_inv(xi1, xi2, xi3, kind, params, 31)
    dfinv_32 = mapping.df_inv(xi1, xi2, xi3, kind, params, 32)
    dfinv_33 = mapping.df_inv(xi1, xi2, xi3, kind, params, 33)
    
    det_df   = mapping.det_df(xi1, xi2, xi3, kind, params)
    
    bx = bx_eq(x, y, z)
    by = by_eq(x, y, z)
    bz = bz_eq(x, y, z)
    
    return (dfinv_31 * bx + dfinv_32 * by + dfinv_33 * bz) * det_df

# equilibrium bulk density (3-form on logical domain)
@types('double','double','double','int','double[:]')
def rho_eq(xi1, xi2, xi3, kind, params):
    
    x = mapping.f(xi1, xi2, xi3, kind, params, 1)
    y = mapping.f(xi1, xi2, xi3, kind, params, 2)
    z = mapping.f(xi1, xi2, xi3, kind, params, 3)
    
    rho_phys = rho_eq_phys(x, y, z)
    
    return rho_phys * mapping.det_df(xi1, xi2, xi3, kind, params)





# =====================================================================
# curl of equilibrium magnetic field (nabla x (DF^T * B_phys)) 

# curl of equilibrium magnetic field (1 - component)
@types('double','double','double','int','double[:]')
def curlb1_eq(xi1, xi2, xi3, kind, params):
    return 0.

# curl of equilibrium magnetic field (2 - component)
@types('double','double','double','int','double[:]')
def curlb2_eq(xi1, xi2, xi3, kind, params):
    return 0.

# curl of equilibrium magnetic field (3 - component)
@types('double','double','double','int','double[:]')
def curlb3_eq(xi1, xi2, xi3, kind, params):
    return 0.

File: hylife/simulation_06042020_2/__pyccel__/equilibrium_MHD.f90
module equilibrium_MHD

use mappings_analytical, only: mapping_tvbypa_det_df => det_df
use mappings_analytical, only: mapping_tvbypa_f => f
use mappings_analytical, only: mapping_tvbypa_df => df
use mappings_analytical, only: mapping_tvbypa_df_inv => df_inv
implicit none




contains

!........................................
function p_eq_phys(x, y, z) result(p_phys)

  implicit none
  real(kind=8)  :: p_phys  
  real(kind=8), value  :: x
  real(kind=8), value  :: y
  real(kind=8), value  :: z



  p_phys = 1.0d0
  !p_phys = (1 - x) * (1 - y) * (1 - z) * x * y * z


  return
end function
!........................................

!........................................
function ux_eq(x, y, z) result(ux)

  implicit none
  real(kind=8)  :: ux  
  real(kind=8), value  :: x
  real(kind=8), value  :: y
  real(kind=8), value  :: z



  ux = 0.0d0
  !ux = cos(2*pi*xi1) * sin(2*pi*xi2) * sin(2*pi*xi3) * 2*pi


  return
end function
!........................................

!........................................
function uy_eq(x, y, z) result(uy)

  implicit none
  real(kind=8)  :: uy  
  real(kind=8), value  :: x
  real(kind=8), value  :: y
  real(kind=8), value  :: z



  uy = 0.0d0
  !uy = sin(2*pi*xi1) * cos(2*pi*xi2) * sin(2*pi*xi3) * 2*pi


  return
end function
!........................................

!........................................
function uz_eq(x, y, z) result(uz)

  implicit none
  real(kind=8)  :: uz  
  real(kind=8), value  :: x
  real(kind=8), value  :: y
  real(kind=8), value  :: z



  uz = 0.0d0
  !uz = sin(2*pi*xi1) * sin(2*pi*xi2) * cos(2*pi*xi3) * 2*pi


  return
end function
!........................................

!........................................
function bx_eq(x, y, z) result(bx)

  implicit none
  real(kind=8)  :: bx  
  real(kind=8), value  :: x
  real(kind=8), value  :: y
  real(kind=8), value  :: z



  bx = 1.0d0
  !bx = sin(2*pi*xi1) * cos(2*pi*xi2) * cos(2*pi*xi3) * (2*pi)**2


  return
end function
!........................................

!........................................
function by_eq(x, y, z) result(by)

  implicit none
  real(kind=8)  :: by  
  real(kind=8), value  :: x
  real(kind=8), value  :: y
  real(kind=8), value  :: z



  by = 1.0d0
  !by = cos(2*pi*xi1) * sin(2*pi*xi2) * cos(2*pi*xi3) * (2*pi)**2


  return
end function
!........................................

!........................................
function bz_eq(x, y, z) result(bz)

  implicit none
  real(kind=8)  :: bz  
  real(kind=8), value  :: x
  real(kind=8), value  :: y
  real(kind=8), value  :: z



  bz = 0.0d0
  !bz = cos(2*pi*xi1) * cos(2*pi*xi2) * sin(2*pi*xi3) * (2*pi)**2


  return
end function
!........................................

!........................................
function rho_eq_phys(x, y, z) result(rho_phys)

  implicit none
  real(kind=8)  :: rho_phys  
  real(kind=8), value  :: x
  real(kind=8), value  :: y
  real(kind=8), value  :: z



  rho_phys = 1.0d0
  !rho_phys = cos(2*pi*xi1) * cos(2*pi*xi2) * cos(2*pi*xi3) * (2*pi)**3


  return
end function
!........................................

!........................................
function p_eq(xi1, xi2, xi3, kind, params) result(p_phys)

  implicit none
  real(kind=8)  :: p_phys  
  real(kind=8), value  :: xi1
  real(kind=8), value  :: xi2
  real(kind=8), value  :: xi3
  integer(kind=8), value  :: kind
  real(kind=8), intent(in)  :: params (0:)
  real(kind=8)  :: x  
  real(kind=8)  :: y  
  real(kind=8)  :: z  



  x = mapping_tvbypa_f(xi1, xi2, xi3, kind, params, 1_8)
  y = mapping_tvbypa_f(xi1, xi2, xi3, kind, params, 2_8)
  z = mapping_tvbypa_f(xi1, xi2, xi3, kind, params, 3_8)


  p_phys = p_eq_phys(x, y, z)


  return
end function
!........................................

!........................................
function u1_eq(xi1, xi2, xi3, kind, params) result(Dummy_8519)

  implicit none
  real(kind=8)  :: Dummy_8519  
  real(kind=8), value  :: xi1
  real(kind=8), value  :: xi2
  real(kind=8), value  :: xi3
  integer(kind=8), value  :: kind
  real(kind=8), intent(in)  :: params (0:)
  real(kind=8)  :: x  
  real(kind=8)  :: y  
  real(kind=8)  :: z  
  real(kind=8)  :: df_11  
  real(kind=8)  :: df_21  
  real(kind=8)  :: df_31  
  real(kind=8)  :: ux  
  real(kind=8)  :: uy  
  real(kind=8)  :: uz  



  x = mapping_tvbypa_f(xi1, xi2, xi3, kind, params, 1_8)
  y = mapping_tvbypa_f(xi1, xi2, xi3, kind, params, 2_8)
  z = mapping_tvbypa_f(xi1, xi2, xi3, kind, params, 3_8)


  df_11 = mapping_tvbypa_df(xi1, xi2, xi3, kind, params, 11_8)
  df_21 = mapping_tvbypa_df(xi1, xi2, xi3, kind, params, 21_8)
  df_31 = mapping_tvbypa_df(xi1, xi2, xi3, kind, params, 31_8)


  ux = ux_eq(x, y, z)
  uy = uy_eq(x, y, z)
  uz = uz_eq(x, y, z)


  Dummy_8519 = df_11*ux + df_21*uy + df_31*uz
  return
end function
!........................................

!........................................
function u2_eq(xi1, xi2, xi3, kind, params) result(Dummy_9151)

  implicit none
  real(kind=8)  :: Dummy_9151  
  real(kind=8), value  :: xi1
  real(kind=8), value  :: xi2
  real(kind=8), value  :: xi3
  integer(kind=8), value  :: kind
  real(kind=8), intent(in)  :: params (0:)
  real(kind=8)  :: x  
  real(kind=8)  :: y  
  real(kind=8)  :: z  
  real(kind=8)  :: df_12  
  real(kind=8)  :: df_22  
  real(kind=8)  :: df_32  
  real(kind=8)  :: ux  
  real(kind=8)  :: uy  
  real(kind=8)  :: uz  



  x = mapping_tvbypa_f(xi1, xi2, xi3, kind, params, 1_8)
  y = mapping_tvbypa_f(xi1, xi2, xi3, kind, params, 2_8)
  z = mapping_tvbypa_f(xi1, xi2, xi3, kind, params, 3_8)


  df_12 = mapping_tvbypa_df(xi1, xi2, xi3, kind, params, 12_8)
  df_22 = mapping_tvbypa_df(xi1, xi2, xi3, kind, params, 22_8)
  df_32 = mapping_tvbypa_df(xi1, xi2, xi3, kind, params, 32_8)


  ux = ux_eq(x, y, z)
  uy = uy_eq(x, y, z)
  uz = uz_eq(x, y, z)


  Dummy_9151 = df_12*ux + df_22*uy + df_32*uz
  return
end function
!........................................

!........................................
function u3_eq(xi1, xi2, xi3, kind, params) result(Dummy_3898)

  implicit none
  real(kind=8)  :: Dummy_3898  
  real(kind=8), value  :: xi1
  real(kind=8), value  :: xi2
  real(kind=8), value  :: xi3
  integer(kind=8), value  :: kind
  real(kind=8), intent(in)  :: params (0:)
  real(kind=8)  :: x  
  real(kind=8)  :: y  
  real(kind=8)  :: z  
  real(kind=8)  :: df_13  
  real(kind=8)  :: df_23  
  real(kind=8)  :: df_33  
  real(kind=8)  :: ux  
  real(kind=8)  :: uy  
  real(kind=8)  :: uz  



  x = mapping_tvbypa_f(xi1, xi2, xi3, kind, params, 1_8)
  y = mapping_tvbypa_f(xi1, xi2, xi3, kind, params, 2_8)
  z = mapping_tvbypa_f(xi1, xi2, xi3, kind, params, 3_8)


  df_13 = mapping_tvbypa_df(xi1, xi2, xi3, kind, params, 13_8)
  df_23 = mapping_tvbypa_df(xi1, xi2, xi3, kind, params, 23_8)
  df_33 = mapping_tvbypa_df(xi1, xi2, xi3, kind, params, 33_8)


  ux = ux_eq(x, y, z)
  uy = uy_eq(x, y, z)
  uz = uz_eq(x, y, z)


  Dummy_3898 = df_13*ux + df_23*uy + df_33*uz
  return
end function
!........................................

!........................................
function b1_eq(xi1, xi2, xi3, kind, params) result(Dummy_9373)

  implicit none
  real(kind=8)  :: Dummy_9373  
  real(kind=8), value  :: xi1
  real(kind=8), value  :: xi2
  real(kind=8), value  :: xi3
  integer(kind=8), value  :: kind
  real(kind=8), intent(in)  :: params (0:)
  real(kind=8)  :: x  
  real(kind=8)  :: y  
  real(kind=8)  :: z  
  real(kind=8)  :: dfinv_11  
  real(kind=8)  :: dfinv_12  
  real(kind=8)  :: dfinv_13  
  real(kind=8)  :: det_df  
  real(kind=8)  :: bx  
  real(kind=8)  :: by  
  real(kind=8)  :: bz  



  x = mapping_tvbypa_f(xi1, xi2, xi3, kind, params, 1_8)
  y = mapping_tvbypa_f(xi1, xi2, xi3, kind, params, 2_8)
  z = mapping_tvbypa_f(xi1, xi2, xi3, kind, params, 3_8)


  dfinv_11 = mapping_tvbypa_df_inv(xi1, xi2, xi3, kind, params, 11_8)
  dfinv_12 = mapping_tvbypa_df_inv(xi1, xi2, xi3, kind, params, 12_8)
  dfinv_13 = mapping_tvbypa_df_inv(xi1, xi2, xi3, kind, params, 13_8)


  det_df = mapping_tvbypa_det_df(xi1, xi2, xi3, kind, params)


  bx = bx_eq(x, y, z)
  by = by_eq(x, y, z)
  bz = bz_eq(x, y, z)


  Dummy_9373 = det_df*(bx*dfinv_11 + by*dfinv_12 + bz*dfinv_13)
  return
end function
!........................................

!........................................
function b2_eq(xi1, xi2, xi3, kind, params) result(Dummy_1603)

  implicit none
  real(kind=8)  :: Dummy_1603  
  real(kind=8), value  :: xi1
  real(kind=8), value  :: xi2
  real(kind=8), value  :: xi3
  integer(kind=8), value  :: kind
  real(kind=8), intent(in)  :: params (0:)
  real(kind=8)  :: x  
  real(kind=8)  :: y  
  real(kind=8)  :: z  
  real(kind=8)  :: dfinv_21  
  real(kind=8)  :: dfinv_22  
  real(kind=8)  :: dfinv_23  
  real(kind=8)  :: det_df  
  real(kind=8)  :: bx  
  real(kind=8)  :: by  
  real(kind=8)  :: bz  



  x = mapping_tvbypa_f(xi1, xi2, xi3, kind, params, 1_8)
  y = mapping_tvbypa_f(xi1, xi2, xi3, kind, params, 2_8)
  z = mapping_tvbypa_f(xi1, xi2, xi3, kind, params, 3_8)


  dfinv_21 = mapping_tvbypa_df_inv(xi1, xi2, xi3, kind, params, 21_8)
  dfinv_22 = mapping_tvbypa_df_inv(xi1, xi2, xi3, kind, params, 22_8)
  dfinv_23 = mapping_tvbypa_df_inv(xi1, xi2, xi3, kind, params, 23_8)


  det_df = mapping_tvbypa_det_df(xi1, xi2, xi3, kind, params)


  bx = bx_eq(x, y, z)
  by = by_eq(x, y, z)
  bz = bz_eq(x, y, z)


  Dummy_1603 = det_df*(bx*dfinv_21 + by*dfinv_22 + bz*dfinv_23)
  return
end function
!........................................

!........................................
function b3_eq(xi1, xi2, xi3, kind, params) result(Dummy_1337)

  implicit none
  real(kind=8)  :: Dummy_1337  
  real(kind=8), value  :: xi1
  real(kind=8), value  :: xi2
  real(kind=8), value  :: xi3
  integer(kind=8), value  :: kind
  real(kind=8), intent(in)  :: params (0:)
  real(kind=8)  :: x  
  real(kind=8)  :: y  
  real(kind=8)  :: z  
  real(kind=8)  :: dfinv_31  
  real(kind=8)  :: dfinv_32  
  real(kind=8)  :: dfinv_33  
  real(kind=8)  :: det_df  
  real(kind=8)  :: bx  
  real(kind=8)  :: by  
  real(kind=8)  :: bz  



  x = mapping_tvbypa_f(xi1, xi2, xi3, kind, params, 1_8)
  y = mapping_tvbypa_f(xi1, xi2, xi3, kind, params, 2_8)
  z = mapping_tvbypa_f(xi1, xi2, xi3, kind, params, 3_8)


  dfinv_31 = mapping_tvbypa_df_inv(xi1, xi2, xi3, kind, params, 31_8)
  dfinv_32 = mapping_tvbypa_df_inv(xi1, xi2, xi3, kind, params, 32_8)
  dfinv_33 = mapping_tvbypa_df_inv(xi1, xi2, xi3, kind, params, 33_8)


  det_df = mapping_tvbypa_det_df(xi1, xi2, xi3, kind, params)


  bx = bx_eq(x, y, z)
  by = by_eq(x, y, z)
  bz = bz_eq(x, y, z)


  Dummy_1337 = det_df*(bx*dfinv_31 + by*dfinv_32 + bz*dfinv_33)
  return
end function
!........................................

!........................................
function rho_eq(xi1, xi2, xi3, kind, params) result(Dummy_9385)

  implicit none
  real(kind=8)  :: Dummy_9385  
  real(kind=8), value  :: xi1
  real(kind=8), value  :: xi2
  real(kind=8), value  :: xi3
  integer(kind=8), value  :: kind
  real(kind=8), intent(in)  :: params (0:)
  real(kind=8)  :: x  
  real(kind=8)  :: y  
  real(kind=8)  :: z  
  real(kind=8)  :: rho_phys  



  x = mapping_tvbypa_f(xi1, xi2, xi3, kind, params, 1_8)
  y = mapping_tvbypa_f(xi1, xi2, xi3, kind, params, 2_8)
  z = mapping_tvbypa_f(xi1, xi2, xi3, kind, params, 3_8)


  rho_phys = rho_eq_phys(x, y, z)


  Dummy_9385 = rho_phys*mapping_tvbypa_det_df(xi1, xi2, xi3, kind, &
      params)
  return
end function
!........................................

!........................................
function curlb1_eq(xi1, xi2, xi3, kind, params) result(Dummy_7817)

  implicit none
  real(kind=8)  :: Dummy_7817  
  real(kind=8), value  :: xi1
  real(kind=8), value  :: xi2
  real(kind=8), value  :: xi3
  integer(kind=8), value  :: kind
  real(kind=8), intent(in)  :: params (0:)

  Dummy_7817 = 0.0d0
  return
end function
!........................................

!........................................
function curlb2_eq(xi1, xi2, xi3, kind, params) result(Dummy_7656)

  implicit none
  real(kind=8)  :: Dummy_7656  
  real(kind=8), value  :: xi1
  real(kind=8), value  :: xi2
  real(kind=8), value  :: xi3
  integer(kind=8), value  :: kind
  real(kind=8), intent(in)  :: params (0:)

  Dummy_7656 = 0.0d0
  return
end function
!........................................

!........................................
function curlb3_eq(xi1, xi2, xi3, kind, params) result(Dummy_7791)

  implicit none
  real(kind=8)  :: Dummy_7791  
  real(kind=8), value  :: xi1
  real(kind=8), value  :: xi2
  real(kind=8), value  :: xi3
  integer(kind=8), value  :: kind
  real(kind=8), intent(in)  :: params (0:)

  Dummy_7791 = 0.0d0
  return
end function
!........................................

end module